arch/arc/core/switch.S

```diff
@@ -132,6 +132,17 @@ _switch_return_from_coop:
 
 	pop_s blink
 #endif /* CONFIG_EXECUTION_BENCHMARKING */
+#ifdef CONFIG_TRACING
```

Comment:
not sure I understand why this change was made.

It seems far better to me to do this in the core kernel, around invocations of `arch_switch()` or `arch_swap()` appropriately.

I really like to avoid doing stuff in arch code if we can do it in the core kernel, x1000 if doing in the arch layer requires assembly!

Reply:
Did you have to do this because of https://github.com/zephyrproject-rtos/zephyr/issues/20671 ?

Reply:
no, I switched trace point to C but this code was left by mistake. fixed now.